### DISCUSSION TITLE
CSP issue between walletConnect and paper.xyz

### DISCUSSION THREAD

**Discussion by devvv3** [WalletConnect/walletconnect-monorepo#3692] [Developer Support] [answered]
Hello,
i'm facing a Content Security Policy issue when trying to use wallet connect with paper.xyz checkoutLink function.

Paper.xyz checkoutLink allow to buy nfts by two ways : one with email connection and one with wallet connection. For this second solution they are using wallet connect web3 modal.

It was working like a charm until last week, but now, impossible to use your modal.

I test it as simple as possible with a simple nextjs13 template project running on localhost:3000 in which i added a simple link to paper checkout link :
```
'use client'
import Image from 'next/image'
import { renderPaperCheckoutLink } from "@paperxyz/js-client-sdk"


export default function Home() {
  const openCheckout = () => {
    renderPaperCheckoutLink({
        checkoutLinkUrl: "https://withpaper.com/checkout/3152e471-95c4-401f-9567-167af40391e1",
    })
}

  return (
    <main className="flex min-h-screen flex-col items-center justify-between p-24">
      
          <button onClick={openCheckout}> click</button>
        
    </main>
  )
}
```
The link open the paper modal and from the paper modal when i'm clicking on "connect a wallet" it should open the walletconnect modal but it don't work and i have the error message : 
```Refused to frame 'https://verify.walletconnect.com/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors https://*.withpaper.com https://withpaper.com"```

So, if i understand well, the issue it's because i'm trying to display walletconnect modal from a domain which is not "https://*.withpaper.com" or "https://withpaper.com"

How can i solve this issue ?

**Comment by glitch-txs** [comment, marked as answer]
This error was fixed on our side, please update you dependencies to their latest versions
